pkg/utilization/utilisation.go

```diff
@@ -0,0 +1,151 @@
+package utilization
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/kubeshop/testkube/pkg/utilization/core"
+
+	"github.com/kubeshop/testkube/cmd/testworkflow-init/output"
+)
+
+const (
+	slowSamplingInterval    = 15 * time.Second
+	fastSamplingInterval    = 1 * time.Second
+	defaultSamplingInterval = fastSamplingInterval
+)
+
+type MetricRecorder struct {
+	writer           core.Writer
+	format           core.Formatter
+	samplingInterval time.Duration
+	tags             []core.KeyValue
+}
+
+type Option func(*MetricRecorder)
+
+func WithFormatter(format core.MetricsFormat) Option {
+	return func(u *MetricRecorder) {
+		formatter, err := core.NewFormatter(format)
+		if err != nil {
+			panic(fmt.Sprintf("failed to create formatter: %v", err))
+		}
+		u.format = formatter
+	}
+}
+
+func WithWriter(writer core.Writer) Option {
+	return func(u *MetricRecorder) {
+		u.writer = writer
+	}
+}
+
+func WithSamplingInterval(interval time.Duration) Option {
+	return func(u *MetricRecorder) {
+		u.samplingInterval = interval
+	}
+}
+
+func WithTags(tags []core.KeyValue) Option {
+	return func(u *MetricRecorder) {
+		u.tags = tags
+	}
+}
+
+func NewMetricsRecorder(opts ...Option) *MetricRecorder {
+	u := &MetricRecorder{
+		format:           core.NewInfluxDBLineProtocolFormatter(),
+		writer:           core.NewSTDOUTWriter(),
+		samplingInterval: defaultSamplingInterval,
+	}
+	for _, opt := range opts {
+		opt(u)
+	}
+	return u
+
+}
+
+func (r *MetricRecorder) Start(ctx context.Context) {
```

Comment:
Likely information about starting/stopping should not be printed (unless debug)

Reply:
yep, all debug logs will be removed